include/boost/asio/buffer.hpp

```diff
@@ -93,9 +93,10 @@ class mutable_buffer
 public:
   /// Construct an empty buffer.
   mutable_buffer() BOOST_ASIO_NOEXCEPT
-    : data_(0),
-      size_(0)
+    : size_(0)
   {
+      static unsigned char c;
+      data_ = &c;
```

Comment:
I don't like the magic static in the constructor body for such trivial use cases. Wouldn't it be nicer to pull `static unsigned char c` out into namespace scope?

Reply:
Technically, this shouldn't be a "magic static." Objects with static storage duration and no explicit initializer are initialized to all-bits-zero before any code executes. This can't be moved to namespace scope because it would make the library no longer header-only. I agree it is a little dirty, but I wanted to give Chris an alternative to just checking `n==0` before calling `memcmp` in case he had a reason for not doing the comparison.